components/sup/src/manager.rs

```diff
@@ -1027,12 +1027,7 @@ impl Manager {
                                             &self.butterfly.service_config_store,
                                             &self.butterfly.service_file_store);
 
-            if self.check_for_changed_services() {
-                self.persist_state_rsr_mlr_gsw();
-            }
-
-            if self.census_ring.changed() {
-                // TODO: combine with above?
+            if self.check_for_changed_services() || self.census_ring.changed() {
```

Comment:
This is fine because `self.census_ring.changed()` is just reporting the value of a boolean, and not performing any additional work, so if this `OR` short-circuits, nothing is missed.